bin/gwd/request.ml

```diff
@@ -349,33 +349,36 @@ let propose_base conf =
   Output.print_sstring conf "</button></li></ul>";
   Hutil.trailer conf
 
-let try_plugin list conf base m =
+let try_plugin list conf base_name m =
   let fn =
     if List.mem "*" list
-    then (fun ( _, fn) -> fn conf base)
-    else (fun (ns, fn) -> (List.mem ns conf.forced_plugins || List.mem ns list) && fn conf base)
+    then (fun ( _, fn) -> fn conf base_name)
+    else (fun (ns, fn) -> (List.mem ns conf.forced_plugins || List.mem ns list) && fn conf base_name)
   in
   List.exists fn (Hashtbl.find_all GwdPlugin.ht m)
 
-let w_lock ~onerror fn conf base =
+let w_lock ~onerror fn conf (base_name : string option) =
   let bfile = Util.bpath (conf.bname ^ ".gwb") in
   Lock.control
-    (Mutil.lock_file bfile) false
-    ~onerror:(fun () -> onerror conf base)
-    (fun () -> fn conf base)
+    (Mutil.lock_file bfile) true
```

Comment:
We wait for the lock instead of only trying and failing.
https://github.com/geneweb/geneweb/blob/d357185658d081842512bb758c7b0ad76072baa8/lib/util/lock.mli#L10